### DISCUSSION TITLE
Support of Python 2.7

### DISCUSSION THREAD

**Discussion by lfarv** [atcollab/at#217] [Python AT]
For how long do we need to support python 2.7?
It's now more that one year that it reached end-of-life…

Dropping 2.7 would allow to use some interesting new constructs.
@willrogers , what's your advice?

**Comment by swhite2401** [comment]
At ESRF we have now fully migrated to python3 so this should not be a problem for us

**Comment by willrogers** [comment]
I think we should drop support for Python 2. It would be tidy to make one more release now before we do it.

The move from Travis CI to Github Actions means that the process of releasing to PyPI will need to be changed. I will do this, but I am very busy at work at the moment and have a baby due in a few weeks!

**Comment by lnadolski** [comment]
At SOLEIL, the migration to Python 3 is not done yet.
Maybe it is the case in other labs
Regards,

Laurent.

**Reply by willrogers** [reply]
@lnadolski you could still use PyAT versions up to 0.1, which I just released. We are considering dropping Python 2.7 on the `master` branch - see https://github.com/atcollab/at/pull/234#issuecomment-795467843

cc @lfarv

**Reply by lfarv** [reply]
Python 2.7 has been marked "End-of-life" more than one year ago! I think it makes sense to stop python 2 support at version 0.1.0. In addition, with virtual environments, it's easy to run AT with python 3 while keeping everything else on an older version.

**Reply by matsievskiysv** [reply]
It might be a good idea add some tag to the last python2 compatible version.
